packages/@uppy/core/types/index.d.ts

```diff
@@ -216,7 +216,7 @@ declare module Uppy {
      * This overload will be removed in Uppy 2.0.
      */
     use(pluginClass: TUseStrictTypes extends StrictTypes ? never : new (uppy: this, opts: any) => Plugin<any>, opts?: object): this
-    getPlugin(name: string): Plugin
+    getPlugin<TPlugin = Plugin>(name: string): TPlugin
```

Comment:
maybe
```suggestion
    getPlugin<TPlugin extends Plugin = Plugin>(name: string): TPlugin
```
so you can't implicitly cast to a totally different type (you could still do a wrong thing like `getPlugin<AwsS3>('Url')` but at least not `getPlugin<number>('Url')`)